### PR TITLE
Api audio GET and PUT endpoints

### PR DESCRIPTION
**Pull request by exterkamp**

### CODE REVIEW

File: packages/api/src/api.ts [hunk 1]
```diff
@@ -85,12 +85,20 @@ export function createApiRouter(localPath: string): express.Router {
...
-  router.get('/messages/:messageId/audio', (req, res) => {
+  router.get('/messages/:messageId/audio', async (req, res) => {
     // Special MessagePaylod that is not json, only bin'd file.
-    res.status(200).end()
+    const audio = await store.getAudio(req.params.messageId)
+    if (!audio) {
+      res.status(404).end()
+    }
+    res.status(200).json(
+      makeResponse<Api.AudioPayload>({audio: {data: audio || ''}}),
```
Comment (exterkamp) [inline packages/api/src/api.ts:95]: @patrickhulce what format will Twilio want?

I doubt `{payload: {audio: {data: 010101010101001}}}` is what they want?

Comment (exterkamp) [reply packages/api/src/api.ts:95]: We could just use express sendFile or something?

File: packages/api/src/storage.ts [hunk 2]
```diff
@@ -120,4 +134,39 @@ export class LocalApiStore {
...
+  async putAudio(messageId: string, audio: Api.AudioFile) {
+    const messages = JSON.parse(fs.readFileSync(this._messagesPath, 'utf8'))
+
+    // Get our message to the messages array.
+    const tmpMessage = messages.messages[messageId]
+    if (!tmpMessage) {
+      return // TODO: error handle
+    }
+    // Set file_path
+    tmpMessage.file_path = tmpMessage.uuid + '.ogg'
```
Comment (exterkamp) [inline packages/api/src/storage.ts:164]: Don't know how we want to handle this internally. I just hardcoded in `ogg`'s as per @connorjclark.

File: packages/api/src/api.ts (hunk 1 above)
Comment (patrickhulce) [reply packages/api/src/api.ts:95]: yeah can just `.send` the buffer itself no need to do JSON

File: packages/api/src/storage.ts [hunk 3]
```diff
@@ -36,7 +39,9 @@ export class LocalApiStore {
 
     const file = JSON.parse(fs.readFileSync(this._messagesPath, 'utf8'))
 
-    return Object.values(file.messages)
+    const msgs = Object.values(file.messages as Api.Message)
```
Comment (patrickhulce) [inline packages/api/src/storage.ts:42]: this works...? but isn't `file.messages: Record<string, Api.Message>`?

File: packages/api/src/storage.ts [hunk 4]
```diff
@@ -120,4 +134,39 @@ export class LocalApiStore {
...
+    if (!message) {
+      return // TODO: error handle
+    }
+    const filePath = message.file_path
+    if (!filePath) {
+      return // TODO: error handle
+    }
+
+    const audio = fs.readFileSync(path.join(this._audioDir, filePath), 'utf8')
+    if (!audio) {
```
Comment (patrickhulce) [inline packages/api/src/storage.ts:149]: not sure this is possible I think it'll just throw file not found

File: packages/api/src/storage.ts (hunk 2 above)
Comment (patrickhulce) [reply packages/api/src/storage.ts:164]: works for me

File: packages/api/src/api.ts (hunk 1 above)
Comment (patrickhulce) [reply packages/api/src/api.ts:95]: I could also see us return base64 in JSON if the requested content type is JSON?

**Pull request merged**